v1/worker.go

```diff
@@ -37,20 +39,49 @@ func (worker *Worker) Launch() error {
 	return <-errorsChan
 }
 
+// RedactURL returns a redacted URL
+func RedactURL(urlstring string) string {
+	u, err := url.Parse(urlstring)
+	if err != nil {
+		log.INFO.Printf("Error while Parsing url", err)
+		return urlstring
+	}
+	if u.User == nil {
+		return urlstring
+	}
+	username := u.User.Username()
+	password, _ := u.User.Password()
+
+	mid := len(username) / 2
+	if len(username)%2 != 0 {
+		mid += 1
+	}
+	username = username[:mid] + strings.Join([]string{strings.Repeat("+", len(username)-mid)}, "")
+
+	mid = len(password) / 2
+	if len(password)%2 != 0 {
+		mid += 1
+	}
+	password = password[:mid] + strings.Join([]string{strings.Repeat("+", len(password)-mid)}, "")
```

Comment:
@yolossn I think password should be completely redacted and should not have length equal to that of the correct password. Knowing the password length greatly reduces computations required for brute force attacks. It should ideally just be 3 asterisks (instead of +).